### ISSUE TITLE
Use Hello.js for OAauth

### ISSUE THREAD

**Issue by RichardLitt** [BeagleLab/beagle#147]

**Comment by RichardLitt** [comment]
Adding a file as a content script doesn't work as it violates isolated world policy - basically, you can't access any variables mentioned in another content script. The only way I can see to get around this is to use event listeners, which is kind of nasty. This attempt won't work well.

**Comment by RichardLitt** [comment]
Due to inline-script security issues, you have to make a separate file in order to have code load in an extension page, hence hello-inject.js. Deferred to make sure one loads after the other. Because you can't have a button with a function in it, either, we had to make an event listener.

However, all of this fails - because the redirect_uri provided to the Google provider won't work, as there is no page to redirect to. Google itself knows about this - they have a crazy hack to make another small redirect file, noted [here](https://groups.google.com/forum/#!topic/oauth2-dev/bmHtIp9ugi0). I don't think we should go down this road, because iirc from the google extension documentation, a Google extension shouldn't actually need to redirect natively -meaning that trying to push hello.js onto this mauy have been suboptimal. For now, this branch is blocked.

**Comment by RichardLitt** [comment]
Working on this here: https://github.com/MrSwitch/hello.js/issues/233#issuecomment-98571896

**Comment by RichardLitt** [comment]
Using Oauth.io. Will check in periodically with issue above.

**Comment by RichardLitt** [comment]
This branch should be able to be unblocked as of https://github.com/MrSwitch/hello.js/issues/233. It's worth trying again.
